src/parseable/streams.rs

```diff
@@ -401,96 +457,345 @@ impl<'a> Stream<'a> {
 
         Schema::try_merge(vec![schema, current_schema]).unwrap()
     }
+
+    pub fn get_first_event(&self) -> Option<String> {
+        self.metadata
+            .read()
+            .expect(LOCK_EXPECT)
+            .first_event_at
+            .clone()
+    }
 }
 
-fn parquet_writer_props(
-    options: &Options,
-    merged_schema: &Schema,
-    time_partition: Option<&String>,
-    custom_partition: Option<&String>,
-) -> WriterPropertiesBuilder {
-    // Determine time partition field
-    let time_partition_field = time_partition.map_or(DEFAULT_TIMESTAMP_KEY, |tp| tp.as_str());
-
-    // Find time partition index
-    let time_partition_idx = merged_schema.index_of(time_partition_field).unwrap_or(0);
-
-    let mut props = WriterProperties::builder()
-        .set_max_row_group_size(options.row_group_size)
-        .set_compression(options.parquet_compression.into())
-        .set_column_encoding(
-            ColumnPath::new(vec![time_partition_field.to_string()]),
-            Encoding::DELTA_BINARY_PACKED,
-        );
+#[derive(Deref, DerefMut, Default)]
+pub struct Streams(RwLock<HashMap<String, StreamRef>>);
+
+// PARSEABLE.streams should be updated
+// 1. During server start up
+// 2. When a new stream is created (make a new entry in the map)
+// 3. When a stream is deleted (remove the entry from the map)
+// 4. When first event is sent to stream (update the schema)
+// 5. When set alert API is called (update the alert)
+impl Streams {
+    pub fn create(
+        &self,
+        options: Arc<Options>,
+        stream_name: String,
+        metadata: LogStreamMetadata,
+        ingestor_id: Option<String>,
+    ) -> StreamRef {
+        let stream = Stream::new(options, &stream_name, metadata, ingestor_id);
+        self.write()
+            .expect(LOCK_EXPECT)
+            .insert(stream_name, stream.clone());
 
-    // Create sorting columns
-    let mut sorting_column_vec = vec![SortingColumn {
-        column_idx: time_partition_idx as i32,
-        descending: true,
-        nulls_first: true,
-    }];
-
-    // Describe custom partition column encodings and sorting
-    if let Some(custom_partition) = custom_partition {
-        for partition in custom_partition.split(',') {
-            if let Ok(idx) = merged_schema.index_of(partition) {
-                let column_path = ColumnPath::new(vec![partition.to_string()]);
-                props = props.set_column_encoding(column_path, Encoding::DELTA_BYTE_ARRAY);
-
-                sorting_column_vec.push(SortingColumn {
-                    column_idx: idx as i32,
-                    descending: true,
-                    nulls_first: true,
-                });
-            }
-        }
+        stream
     }
 
-    // Set sorting columns
-    props.set_sorting_columns(Some(sorting_column_vec))
-}
+    pub fn delete_stream(&self, stream_name: &str) {
+        self.write().unwrap().remove(stream_name);
+    }
```

Comment:
```suggestion
    pub fn delete(&self, stream_name: &str) {
        self.write().expect(LOCK_EXPECT).remove(stream_name);
    }
```

update caller accordingly